pkgs/development/python-modules/cartopy/default.nix

```
@@ -0,0 +1,45 @@
+{ buildPythonPackage, lib, fetchPypi,
+  pytest, filelock, mock, pep8,
+  cython, isPy37,
+  six, pyshp, shapely, geos, proj, numpy,
+  gdal, pillow, matplotlib, pyepsg, pykdtree, scipy, owslib, fiona
+}:
+
+buildPythonPackage rec {
+  pname = "cartopy";
+  version = "0.17.0";
+
+  src = fetchPypi {
+    inherit version;
+    pname = "Cartopy";
+    sha256 = "0q9ckfi37cxj7jwnqnzij62vwcf4krccx576vv5lhvpgvplxjjs2";
+  };
+
+  checkInputs = [ filelock mock pytest pep8 ];
+
+  # several tests require network connectivity: we disable them
+  checkPhase = ''
+    export HOME=$(mktemp -d)
+    pytest --pyargs cartopy \
+      -m "not network and not natural_earth" \
+      -k "not test_nightshade_image"
+  '';
+
+  buildInputs = [ cython ];
+
+  propagatedBuildInputs = [
+    # required
+    six pyshp shapely geos proj numpy
+
+    # optional
+    gdal pillow matplotlib pyepsg pykdtree scipy fiona
+  ] ++ lib.optionals (!isPy37) [ owslib ]; # owslib is broken with python3.7
+
+  meta = with lib; {
+    description = ''Process geospatial data to create maps and perform analyses.'';
```

```suggestion
    description = "Process geospatial data to create maps and perform analyses"';
```
Double quotes are fine here, and we'd don't use periods at the end of the description.

---

Done commas, dot and quotes